store/src/java/com/zimbra/cs/service/mail/CopyToFiles.java

```diff
@@ -55,120 +54,183 @@ public CopyToFiles(
     // get attachment name
-    final Try<String> fileNameTry = attachmentTry.mapTry(attachment -> attachment.getFileName())
-        .onFailure(ex -> mLog.debug(ex.getMessage()))
-        .mapFailure(Case($(instanceOf(Exception.class)),
-            ex -> ServiceException.FAILURE("Cannot get file name.", ex)));
+    final Try<String> fileNameTry = attachmentTry.flatMapTry(this::getAttachmentName);
     // get attachment content stream
-    final Try<InputStream> attachmentStreamTry = attachmentTry.mapTry(Part::getInputStream)
-        .onFailure(ex -> mLog.debug(ex.getMessage()));
+    final Try<InputStream> attachmentStreamTry =
+        attachmentTry.mapTry(Part::getInputStream).onFailure(ex -> mLog.debug(ex.getMessage()));
 
-    // execute Files api call
-    String nodeId = Optional.ofNullable(
-            For(authCookieTry, destFolderIdTry, attachmentStreamTry, attachmentSizeTry, fileNameTry,
-                    contentTypeTry)
-                .yield(
-                    (authCookie, destFolderId, attachmentStream, attachmentSize, fileName, contentType) ->
-                        filesClient.uploadFile(authCookie, destFolderId, fileName, contentType,
+    // execute Files api call and flatten yield result
+    // maps every non-ServiceException to internal error
+    String nodeId =
+        Optional.ofNullable(
+                For(
+                        authCookieTry,
+                        destFolderIdTry,
+                        fileNameTry,
+                        contentTypeTry,
+                        attachmentStreamTry,
+                        attachmentSizeTry)
+                    .yield(
+                        (authCookie,
+                            destFolderId,
+                            fileName,
+                            contentType,
                             attachmentStream,
-                            attachmentSize).get())
-                .onFailure(ex -> mLog.debug(ex.getMessage()))
-                .mapFailure(Case($(ex -> !(ex instanceof ServiceException)),
-                    ex -> ServiceException.FAILURE("internal error.", ex)))
-                .get())
-        .orElseThrow(() -> ServiceException.FAILURE("got null response from Files server."))
-        .getNodeId();
+                            attachmentSize) ->
+                            filesClient.uploadFile(
+                                authCookie,
+                                destFolderId,
+                                fileName,
+                                contentType,
+                                attachmentStream,
+                                attachmentSize))
+                    .flatMap(x -> x)
+                    .onFailure(ex -> mLog.debug(ex.getMessage()))
+                    .mapFailure(
+                        Case(
+                            $(ex -> !(ex instanceof ServiceException)),
+                            ex -> ServiceException.FAILURE("internal error.", ex)))
```

Comment:
You can use `onFailure` method to avoid `instanceOf` that is not for being slow.
```suggestion
                    .onFailure(ServiceException.class, ex -> ServiceException.FAILURE("internal error.", ex))
```